clients/src/main/java/org/apache/kafka/clients/consumer/internals/ShareFetchMetricsManager.java

```diff
@@ -0,0 +1,118 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.kafka.clients.consumer.internals;
+
+import org.apache.kafka.common.MetricName;
+import org.apache.kafka.common.metrics.Metrics;
+import org.apache.kafka.common.metrics.Sensor;
+import org.apache.kafka.common.metrics.stats.Avg;
+import org.apache.kafka.common.metrics.stats.Max;
+import org.apache.kafka.common.metrics.stats.Meter;
+import org.apache.kafka.common.metrics.stats.WindowedCount;
+
+import static org.apache.kafka.clients.consumer.internals.ConsumerUtils.SHARE_CONSUMER_METRIC_GROUP_PREFIX;
+
+public class ShareFetchMetricsManager {
+    private final FetchMetricsRegistry metricsRegistry;
+    private final Sensor throttleTime;
+    private final Sensor bytesFetched;
+    private final Sensor recordsFetched;
+    private final Sensor fetchLatency;
+    private final Sensor sentAcknowledgements;
+    private final Sensor failedAcknowledgements;
+    public final MetricName acknowledgementSendRate;
+    public final MetricName acknowledgementSendTotal;
+    public final MetricName acknowledgementErrorRate;
+    public final MetricName acknowledgementErrorTotal;
+    public final MetricName recordsFetchedRate;
+    public final MetricName recordsFetchedTotal;
+    public final MetricName recordsPerRequestAvg;
+    public final MetricName recordsPerRequestMax;
+
+    public ShareFetchMetricsManager(Metrics metrics, FetchMetricsRegistry fetchMetricsRegistry) {
```

Comment:
Why did you not make a ShareFetchMetricsRegistry?

Reply:
I had initially thought so as lot of metrics were already present in the existing FetchMetricsRegistry. But yeah, as we had discussed on-call, this is a small piece of code and the design would be cleaner with more metrics in future if we have a separate ShareFetchMetricsRegistry. Have made the changes.